target/scripts/startup/setup.d/mail_state.sh

```diff
@@ -3,123 +3,126 @@
+  # Notably avoids copying over 200MB for the ClamAV database
+  [[ ${ENABLE_AMAVIS}       -eq 1 ]] && SERVICEDIRS+=('lib/amavis')
+  [[ ${ENABLE_CLAMAV}       -eq 1 ]] && SERVICEDIRS+=('lib/clamav')
+  [[ ${ENABLE_FAIL2BAN}     -eq 1 ]] && SERVICEDIRS+=('lib/fail2ban')
+  [[ ${ENABLE_FETCHMAIL}    -eq 1 ]] && SERVICEDIRS+=('lib/fetchmail')
+  [[ ${ENABLE_GETMAIL}      -eq 1 ]] && SERVICEDIRS+=('lib/getmail')
+  [[ ${ENABLE_MTA_STS}      -eq 1 ]] && SERVICEDIRS+=('lib/mta-sts')
+  [[ ${ENABLE_POSTGREY}     -eq 1 ]] && SERVICEDIRS+=('lib/postgrey')
+  [[ ${ENABLE_RSPAMD}       -eq 1 ]] && SERVICEDIRS+=('lib/rspamd')
+  [[ ${ENABLE_RSPAMD_REDIS} -eq 1 ]] && SERVICEDIRS+=('lib/redis')
+  [[ ${ENABLE_SPAMASSASSIN} -eq 1 ]] && SERVICEDIRS+=('lib/spamassassin')
+  [[ ${ENABLE_SRS}          -eq 1 ]] && SERVICEDIRS+=('lib/postsrsd')
+  [[ ${SMTP_ONLY}           -ne 1 ]] && SERVICEDIRS+=('lib/dovecot')
+
+  # Single service files
+  [[ ${ENABLE_SRS}          -eq 1 ]] && SERVICEFILES+=('/etc/postsrsd.secret')
+
+  for SERVICEFILE in "${SERVICEFILES[@]}"; do
+    DEST="${DMS_STATE_DIR}/${SERVICEFILE}"
+    DESTDIR="${DEST%/*}"
+
+    mkdir -p "${DESTDIR}"
+    if [[ -f ${DEST} ]]; then
+      _log 'trace' "Destination ${DEST} exists, linking ${SERVICEFILE} to it"
+      # Original content from image no longer relevant, remove it:
+      rm -f "${SERVICEFILE}"
+    elif [[ -f "${SERVICEFILE}" ]]; then
+      _log 'trace' "Moving ${SERVICEFILE} to ${DEST}"
+      # Empty volume was mounted, or new content from enabling a feature ENV:
+      mv "${SERVICEFILE}" "${DEST}"
+      # Apply SELinux security context to match the state directory, so access
+      # is not restricted to the current running container:
+      chcon -R --reference="${DMS_STATE_DIR}" "${DEST}" 2>/dev/null || true
+    fi
+
+    # Symlink the original file in the container ($SERVICEFILE) to be
+    # sourced from assocaiated path in /var/mail-state/ ($DEST):
+    ln -s "${DEST}" "${SERVICEFILE}"
+  done
+
+  for SERVICEDIR in "${SERVICEDIRS[@]}"; do
+    DEST="${DMS_STATE_DIR}/${SERVICEDIR//\//-}"
+    SERVICEDIR="/var/${SERVICEDIR}"
+
+    # If relevant content is found in /var/mail-state (presumably a volume mount),
+    # use it instead. Otherwise copy over any missing directories checked.
+    if [[ -d ${DEST} ]]; then
+      _log 'trace' "Destination ${DEST} exists, linking ${SERVICEDIR} to it"
+      # Original content from image no longer relevant, remove it:
+      rm -rf "${SERVICEDIR}"
+    elif [[ -d ${SERVICEDIR} ]]; then
+      _log 'trace' "Moving contents of ${SERVICEDIR} to ${DEST}"
+      # An empty volume was mounted, or new content dir now exists from enabling a feature ENV:
+      mv "${SERVICEDIR}" "${DEST}"
+      # Apply SELinux security context to match the state directory, so access
+      # is not restricted to the current running container:
+      chcon -R --reference="${DMS_STATE_DIR}" "${DEST}" 2>/dev/null || true
```

Comment:
Inlining a reference since this PR will make `git blame` a lil annoying to find it otherwise 😅 (_I didn't suggest this for the earlier duplicate comment, since that's just a single file and for the most part it looks like this logic could be DRY as a function call if someone wants to refactor_).

```suggestion
      # Apply SELinux security context to match the state directory, so access
      # is not restricted to the current running container:
      # https://github.com/docker-mailserver/docker-mailserver/pull/3890
      chcon -R --reference="${DMS_STATE_DIR}" "${DEST}" 2>/dev/null || true
```